### DISCUSSION TITLE
Implementing a user based singleton collection

### DISCUSSION THREAD

**Discussion by laekem34** [directus/directus#10614] [Feature Requests]
Hello !

I think it could be a good idea to have possibilty to create a singleton which content is different between users.

Actually, when we create a singleton, every users (if they have permissions) access to the same data, a global singleton.

It could be interesting to have a custmizable specific singleton (which could be represented by a O2O relation with user) and every singleton will be a row in table

An use case could be if we have an api to configure a member parameter's for other application. So not to weigh down the user account config page, he could have a dedicated configuration page in content module like a singleton but with data linked to user.

Thanks you :)

**Comment by rijkvanzanten** [comment]
I think we could/should base this on the regular permissions configuration of the user 🤔 

That way, if the role has full read access to the collection, the app/api will just use the item with id = `1`. If the role/user can't read that particular item, the app/api will automatically create a new one (with id = `2`) in this case.

That way, you could easily configure it with a user-scope by adding a m2o field from the singleton to the user, and setting the permissions to `{ user: { _eq: $CURRENT_USER }}`

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
